### DISCUSSION TITLE
How to verify a private blockchain integrity using Geth or Truffle?

### DISCUSSION THREAD

**Discussion by luizhstorres** [ConsenSys-archive/truffle#5733] [General Q&A] [answered]
Hello. I studied about Geth and Truffle. I still consider myself new to this technology. How to verify a private blockchain integrity? How to know if a transaction or block was tampered? Is there any inherent function on Geth or Truffle?
Case Study: Assuming that data from a spreadsheet is periodically stored (uploaded) into blocks on a private blockchain. If someone accidentally or maliciously makes a change to old data in the spreadsheet, which has already been uploaded to the blockchain, how to know if the data has changed? Is there any function or mechanism inherent developed on Geth to check if the information (e.g.: the block hash) of any block in the entire chain has changed? How to audit a private blockchain for possible changes? Thank you for helping!

**Comment by apdan510** [comment, marked as answer]
@luizhstorres Typically one will compare your current node and logs versus a fully confirmed and accepted block on a blockchain explorer index like Etherscan. 

In your example when the user uploads data onto the blockchain after it is confirmed by multiple nodes and accepted onto the chain. The data from the spreadsheet is stored into the network and cannot be modified once on chain. If data from the old spreadsheet has been modified, you would have to manually verify the data between the local data from your spreadsheet and what is stored onchain.  

I do not think Truffle has a tool to compare data like in your example. However, Consensys does offer multiple security and auditing services here at https://consensys.net/diligence/ if you want a professional team to audit, analyze, or verify.
